### DISCUSSION TITLE
How to merge two trained model into one model?

### DISCUSSION THREAD

**Discussion by sigit-purnomo** [explosion/spaCy#12997] [Help: Best practices]
Hi all,

I have trained a POS Tagging and Dependency model using the spaCy project template and made the model into a package. I have also trained a model for NER (only the NER component) using a different corpus/dataset and made it into a package.

I have read this [thread](https://github.com/explosion/spaCy/discussions/3056?sort=new), but there are many ways. What's the best practice to merge both models into one package?

Thank you

**Comment by danieldk** [comment]
The standard way we do this is using the spaCy `assemble` subcommand. `assemble` allows you to create a spaCy pipeline without training. You can source the components from the pipelines that you already created. You can find an example of sourcing in the coref example project:

https://github.com/explosion/projects/blob/e24a085669b4db6918ffeb2752846089d8dee57a/experimental/coref/configs/coref.cfg#L28

You can then call `spacy assemble` on the configuration, see the same project for an [example](https://github.com/explosion/projects/blob/e24a085669b4db6918ffeb2752846089d8dee57a/experimental/coref/project.yml#L151).

**Reply by sigit-purnomo** [reply]
Thank you Daniel
